### PR TITLE
feat(query): system.columns filter view engine fields

### LINKED ISSUE #10632: bug: information_schema.column and system.columns will generate recursion
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

main version

### What's Wrong?

information_schema.columns is a view that will query system.columns

if query information_schema.columns will call dump_table_columns and information_schema also in CATALOG_DEFAULT, will recursion

### How to Reproduce?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

system.columns filter view engine fields

Closes #10632

### CODE REVIEW

**Review by sundy-li — APPROVED** [review]

File: src/query/storages/system/src/columns_table.rs [hunk 1]
```diff
@@ -144,16 +144,18 @@ impl ColumnsTable {
...
-                        let mut planner = Planner::new(ctx.clone());
-                        let (plan, _) = planner.plan_sql(query).await?;
-                        let schema = infer_table_schema(&plan.schema())?;
-                        schema.fields().clone()
-                    } else {
-                        return Err(ErrorCode::Internal(
-                            "Logical error, View Table must have a SelectQuery inside.",
-                        ));
-                    }
+                    // information_schema.columns is a view that will query system.columns
```
Comment (BohuTANG) [inline src/query/storages/system/src/columns_table.rs:147]: Is this a temporary fix?

Comment (BohuTANG) [reply src/query/storages/system/src/columns_table.rs:147]: Suggest to add more comment and remove the unused code

Comment (TCeason) [reply src/query/storages/system/src/columns_table.rs:147]: Ok

Comment (BohuTANG) [reply src/query/storages/system/src/columns_table.rs:147]: It's better to link the issue here.

**Pull request merged**